bundles/org.openhab.binding.sagercaster/src/main/java/org/openhab/binding/sagercaster/internal/SagerWeatherCaster.java

```diff
@@ -0,0 +1,5374 @@
+ *
+ *          Copyright © 2008 Naish666 (eltiempo.selfip.com)
+ *               October 2008 - v1.0
+ *          Java transposition done by Gaël L'hopital - 2015
+ **
+ *     This program is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    This program is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ ***
+ * BT's Global Sager Weathercaster PHP Scripts For Cumulus (Weathercaster)
+ * by "Buford T. Justice" / "BTJustice"
+ * http://www.freewebs.com/btjustice/bt-forecasters.html
+ * 2014-02-05
+ *
+ * You may redistribute and use these PHP Scripts any way you wish as long as
+ * they remain FREE and money is not charged for their use directly or indirectly.
+ * If these PHP Scripts are used in your work or are modified in any way, please
+ * retain the full credit header.
+ * Based Upon:
+ * The Sager Weathercaster:  A Scientific Instrument for Accurate Prediction of
+ * the Weather
+ * Copyright © 1969 by Raymond M. Sager and E. F. Sager
+ " The Sager Weathercaster predicts the weather quickly and accurately.  It has been
+ * in use since 1942.
+ * Not a novelty, not a toy, this is a highly dependable, scientifically designed
+ * tool of inestimable value to travelers, farmers, hunters, sailors, yachtsmen, campers,
+ * fishermen, students -- in fact, to everyone who needs or wants to know what
+ * the weather will be."
+ * 378 possible forecasts determined from 4996 dial codes.
+ */
+
+package org.openhab.binding.sagercaster.internal;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This class is responsible for handling the SagerWeatherCaster algorithm
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+public class SagerWeatherCaster {
+
+    // Northern Polar Zone & Northern Tropical Zone
+    private final static String[] NPZDIRECTIONS = { "S", "SW", "W", "NW", "N", "NE", "E", "SE" };
+    // Northern Temperate Zone
+    private final static String[] NTZDIRECTIONS = { "N", "NE", "E", "SE", "S", "SW", "W", "NW" };
+    // Southern Polar Zone & Southern Tropical Zone
+    private final static String[] SPZDIRECTIONS = { "N", "NW", "W", "SW", "S", "SE", "E", "NE" };
+    // Southern Temperate Zone
+    private final static String[] STZDIRECTIONS = { "S", "SE", "E", "NE", "N", "NW", "W", "SW" };
+
+    private final static Map<String, String> sageForecaster = new HashMap<String, String>() {
```

Comment:
This map is unreasonably large to be populated directly in source code. I would suggest putting the data in another file, perhaps a .properties file, and then loading the file at runtime.